cg/apps/hermes/models.py

```diff
@@ -16,23 +17,24 @@ class CGTag(BaseModel):
     mandatory: Optional[bool] = False
 
 
+def check_mandatory_exists(file: CGTag):
+    if file.mandatory:
+        assert Path(file.path).exists(), f"Mandatory file cannot be found at {file.path}"
+        return file
+    if not Path(file.path).exists():
+        LOG.info("Optional file %s not found, removing from bundle", file.path)
+        return
+    return file
+
+
 class CGDeliverables(BaseModel):
     """Class that specifies the output from hermes"""
 
     pipeline: str
     bundle_id: str
-    files: List[CGTag]
-
-    @validator("files", each_item=True)
-    def check_mandatory_exists(cls, file):
-        if file.mandatory:
-            assert Path(file.path).exists(), f"Mandatory file cannot be found at {file.path}"
-            return file
-        if not Path(file.path).exists():
-            LOG.info("Optional file %s not found, removing from bundle", file.path)
-            return
-        return file
+    files: List[Annotated[CGTag, AfterValidator(check_mandatory_exists)]]
 
-    @validator("files")
+    @classmethod
+    @field_validator("files")
     def remove_invalid(cls, value):
         return [item for item in value if item]
```

Comment:
Not strictly part of the Pydantic migration, but could you take the opportunity to improve the function name, signature and use more mnemonic variables

Reply:
Sure thing!

Reply:
I have tried reworking this to a more understandable validation process now. Can you see if the current validation matches the old one? @henrikstranneheim